Inventory/MongoDbInventory.cs

```diff
@@ -0,0 +1,81 @@
+﻿using InventorySystem.DatabseContext;
+using InventorySystem.Models;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace InventorySystem.Inventory;
+
+public class MongoDbInventory : IInventory
+{
+    private readonly MongoDbContext _dbContext;
+
+    public MongoDbInventory()
+    {
+        _dbContext = new();
+    }
+
+    public bool Exists(string name)
+    {
+        var documents = _dbContext.Products.Find(new BsonDocument()).ToList();
```

Comment:
instead of loading the whole document, you can do in-db filtering and just try to load the needed obj.

Reply:
ok, I will try that